### PR TITLE
JC-2400 Error with user in several groups correction

### PR DESCRIPTION
**Pull request by usharik**

### CODE REVIEW

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java [hunk 1]
```diff
@@ -666,12 +671,12 @@ public void testFindByEmailNotInGroupId() {
...
-    @Test
+    @Test(invocationCount = 5)
     public void testFindUser_IfHeHasNoGroups() {
-        final String keyWord = "keyword";
+        Random rnd = new Random();
+        final String keyWord = randomAlphanumeric(7);;
         Group group1 = createGroup("testGroup1");
         JCUser jcUser;
-        Random rnd = new Random();
         if (rnd.nextBoolean()) {
```
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: You can just use `bool()` from datagen

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java [hunk 2]
```diff
@@ -666,12 +671,12 @@ public void testFindByEmailNotInGroupId() {
         Assert.assertEquals(jcUser1group1.getUsername(), result.get(0).getUsername());
     }
 
-    @Test
+    @Test(invocationCount = 5)
     public void testFindUser_IfHeHasNoGroups() {
```
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Just `findsUser_ifHeIsNotInAnyGroup`

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java [hunk 3]
```diff
@@ -666,12 +671,12 @@ public void testFindByEmailNotInGroupId() {
         Assert.assertEquals(jcUser1group1.getUsername(), result.get(0).getUsername());
     }
 
-    @Test
+    @Test(invocationCount = 5)
```
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: No big need for the invocation count actually..

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java (hunk 1 above)
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Um.. even better:

```
jcUser = createUserWithMail(sample("username", keyword.toUpperCase()), sample("m@m.a", keyword.toUpperCase(), true);
```

This is shorter and will test more scenarios.

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java [hunk 4]
```diff
@@ -685,37 +690,45 @@ public void testFindUser_IfHeHasNoGroups() {
 
     @Test(invocationCount = 5)
     public void testDontFindUser_IfHeIsInGroup() {
```
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: `doesNotFindUser_ifHeIsInGroup()`

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java [hunk 5]
```diff
@@ -685,37 +690,45 @@ public void testFindUser_IfHeHasNoGroups() {
...
-        final String keyWord = "keyword";
-        Group group1 = createGroup("testGroup1");
         Random rnd = new Random();
+        final String keyWord = randomAlphanumeric(7);;
+        List<Group> grpList;
+        JCUser user;
         if (rnd.nextBoolean()) {
-            createUserWithGroup(keyWord, "mail@mail.ru", group1);
+            user = createUserWithGroup(keyWord, "mail@mail.ru", createGroup("testGroup0"));
```
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Same logic with `sample()` can go here.

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java [hunk 6]
```diff
@@ -685,37 +690,45 @@ public void testFindUser_IfHeHasNoGroups() {
...
         } else {
-            createUserWithGroup("username", keyWord + "@mail.ru", group1);
+            user = createUserWithGroup("username", keyWord + "@mail.ru", createGroup("testGroup0"));
+        }
+        grpList = addRandomGroupsToUser(user);
+        for (Group group : grpList) {
```
Comment (ctapobep) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]:
```
List<UserDto> result = userDao.findByUsernameOrEmailNotInGroup(keyWord, sample(grpList).getId(), 20);
assertEquals(result.size(), 0);
```

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java (hunk 1 above)
Comment (usharik) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Done

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java (hunk 2 above)
Comment (usharik) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Done

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java (hunk 4 above)
Comment (usharik) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Done

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java (hunk 5 above)
Comment (usharik) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: Done

File: jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java (hunk 6 above)
Comment (usharik) [inline jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java]: From my point of view If we are checking for all added group we provide better test coverage.

**Pull request merged**